### DISCUSSION TITLE
Pre-requisites for MQB sections/chapters

### DISCUSSION THREAD

**Discussion by mhasoba** [mhasoba/TheMulQuaBio#147] [General]
Hello all, and in particular @yanzhu624,  @2piruben, @tinyuhui, @vaajansen, and @BhavKhatri , we need to add pre-requisites for certain chapters.

For example, @willpearse has provided the following for his Machine Learning week (albeit not currently a MQB chapter):

* What a likelihood space is. I realise they are meant to have done that in the ML week but every time when I draw two parameters and then say "OK, the 'height' of this space is the likelihood - just like in an ML search" every year they pull a face and I say "you did this in the ML week, right?" and they all look a bit sheepish and say "yes, but we don't understand it". I then teach it to them conceptually, which is fine, but that's a dependency.

* Knowing what it means to minimise a function, and ideally subject to some conditions. When I say things like "we are minimising the sum of the squared errors, searching across a space of parameters" they're often lost. Which is fine - I go through what an optimisation means again - but it's notable to me because I am certain that you teach them this!

* What an eigenvector and eigvenvalue are. They should know, I know, but every year they are a little shaky...

* The ability to spot-check difficult R/Python installations. Every year I ask them to install keras/TensorFlow and every year there is at least one student who isn't on Linux (and, god help me, they're normally on Windows) and needs help with rather mundane things. 

Please list your chapter/week's prerequsites in this discussion thread.

**Comment by mhasoba** [comment]
@2piruben, @tinyuhui,  @BhavKhatri , and @vaajansen , gentle reminder...

**Comment by vaajansen** [comment]
Samraat, you were going to merging the proposed structure. That was your last message before the break. Until we know who is doing what and when we can't do much.

From: mhasoba ***@***.***>
Sent: Monday, January 6, 2025 4:56 PM
To: MulQuaBio/MQB ***@***.***>
Cc: Jansen, Vincent ***@***.***>; Mention ***@***.***>
Subject: [EXT] Re: [MulQuaBio/MQB] Pre-requisites for MQB sections/chapters (Discussion #147)


@2piruben<https://github.com/2piruben>, @tinyuhui<https://github.com/tinyuhui>, @BhavKhatri<https://github.com/BhavKhatri> , and @vaajansen<https://github.com/vaajansen> , gentle reminder...

-
Reply to this email directly, view it on GitHub<https://github.com/orgs/MulQuaBio/discussions/147#discussioncomment-11751423>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BNHZQ6RLBBMOBXFRWAE7VPT2JKYSZAVCNFSM6AAAAABTN6QD7SVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNZVGE2DEMY>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

This email, its contents and any attachments are intended solely for the addressee and may contain confidential information. In certain circumstances, it may also be subject to legal privilege. Any unauthorised use, disclosure, or copying is not permitted. If you have received this email in error, please notify us and immediately and permanently delete it. Any views or opinions expressed in personal emails are solely those of the author and do not necessarily represent those of Royal Holloway, University of London. It is your responsibility to ensure that this email and any attachments are virus free.

**Reply by mhasoba** [reply]
Just to clarify - all I am  asking for is a list of maths/computing pre-requsites for your taught content to be listed in this thread...

**Reply by mhasoba** [reply]
along the lines of the list provided by Will Pearse at the start of this thread

**Comment by vaajansen** [comment]
Sorry, didn't get that. Here goes


  *   First of all, the ability to manipulate equations and calculate equilibria of systems of differential/difference eqns
  *   Functions, being able to plot functions as function of arguments. Plotting functions in 2D and 3D
  *   Being able do differentiation (whilst standing on their heads). Knowledge of exponential functions, logarithms, logistic functions, trigonometric functions.
  *   Basic knowledge polynomials and understanding linear/non linear
  *   Integration (at least the concept)
  *   Matrix multiplication. Eigenvalues, eigenvectors.
  *   Complex numbers and relation between exponential and trigonometric functions through complex numbers
  *   Differential equations. General solutions of systems of linear ODEs. Difference equations. Relation to eigenvalues and multipliers. Characteristic eqn
  *   Phase plane representation of solutions of ODEs/difference eqns
  *   Routh Hurwitz criteria for 2D systems

The bulk of that is A level stuff (but not the linear algebra, possibly complex numbers, linear ODEs). Much of this is spelled out in text books or wikipedia. Check eg Otto and Day.

From: mhasoba ***@***.***>
Sent: Monday, January 6, 2025 5:05 PM
To: MulQuaBio/MQB ***@***.***>
Cc: Jansen, Vincent ***@***.***>; Mention ***@***.***>
Subject: [EXT] Re: [MulQuaBio/MQB] Pre-requisites for MQB sections/chapters (Discussion #147)


along the lines of the list provided by Will Pearse at the start of this thread

-
Reply to this email directly, view it on GitHub<https://github.com/orgs/MulQuaBio/discussions/147#discussioncomment-11751539>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BNHZQ6TULP4AR4PBX4TBWST2JKZSXAVCNFSM6AAAAABTN6QD7SVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNZVGE2TGOI>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

This email, its contents and any attachments are intended solely for the addressee and may contain confidential information. In certain circumstances, it may also be subject to legal privilege. Any unauthorised use, disclosure, or copying is not permitted. If you have received this email in error, please notify us and immediately and permanently delete it. Any views or opinions expressed in personal emails are solely those of the author and do not necessarily represent those of Royal Holloway, University of London. It is your responsibility to ensure that this email and any attachments are virus free.

**Reply by BhavKhatri** [reply]
For mine:

- Taylor series, limits

- basic differentiation and integration

- 1st order differential equations (exponential growth, Logistic, Integrating factor method)

- Random variables and basic probability concepts like expectation of a random variable

- Binomial distribution (mean np and variance np*(1-p))

- Poisson distribution (mean np variance np)

**Comment by 2piruben** [comment]
Here are mine:

- Manipulating probability distributions including joint, marginal, and conditional probabilities
- Familiarity with key distributions (e.g., Gaussian, Binomial, Poisson)
- Derivation of maximum likelihood estimates for parameter inference
- Understanding frequentist hypothesis testing, including null/alternative hypotheses and interpreting p-values.
- Setting up and analyzing first order ordinary differential equations





Dr. Ruben Perez-Carrasco (he/him)

Senior Lecturer in Theoretical Systems Biology

Department of Life Sciences

Imperial College London,  SW7 2AZ, UK



http://www.perez-carrasco.com/


My working hours may not coincide with yours. I do not expect you to reply to this email outside your working hours.

________________________________
